synapse/http/matrixfederationclient.py

```diff
@@ -479,6 +480,14 @@ async def _send_request(
             RequestSendFailed: If there were problems connecting to the
                 remote, due to e.g. DNS failures, connection timeouts etc.
         """
+        # Validate server name and log if it is an invalid destination, this is
+        # partially to help track down code paths where we haven't validated before here
+        try:
+            parse_and_validate_server_name(request.destination)
+        except ValueError:
+            logger.info(f"Invalid destination: {request.destination}.")
```

Comment:
We might want to do a `logger.exception()` or `logger.error(..., stack_info=True)`, so that the error shows up in sentry with a stack trace. There is a chance that the exception we raise might be caught and silenced.

Reply:
Great I have done so.

Reply:
After poking around in sentry, it looks like sentry may truncate the stack from `stack_info=True`: https://sentry.matrix.org/sentry/synapse-matrixorg/issues/15795

I think we should use `logger.exception` instead:
```py
    logger.exception(f"Invalid destination: {request.destination}.")
```
which will produce something like: https://sentry.matrix.org/sentry/synapse-matrixorg/issues/258079/

Sorry about that!

Reply:
No worries these are helpful things for me to know to consider!